src/libs/export.c

```diff
@@ -213,6 +227,135 @@ void _set_dimensions(dt_lib_export_t *d, int max_width, int max_height)
   g_free(max_height_char);
 }
 
+static void scale_changed(GtkEntry *spin)
+{
+  const char *validSign = ",.0123456789";
+  const gchar *value = "";
+  value =  gtk_entry_get_text(spin);
+
+  int len = sizeof(value);
+  int i, j = 0, idec = 0, idiv = 0, pdiv = 0;
+  char new_value[30] = "";
+
+  for (i = 0; i < len; i++) {
+    char *val = strchr(validSign, value[i]);
+    if (val == NULL) {
+      if (idiv==0) {
+        if (i == 0) {
+          new_value[j++] = '1';
+        } else {
+          if (atof(value) == 0.0) {
+            new_value[0] = '1';
+          }
+          idec = 0;
+          idiv = 1;
+          new_value[j++] = '/';
+          pdiv = j;
+        }
+      }
+    } else if ((val[0] == '.') || (val[0] == ',')) {
+      if (idec == 0) {
+        if ((i == 0) || (i == pdiv)) {
+          new_value[j++] = '0';
+        } else {
+          idec = 1;
+          new_value[j++] = value[i];
+        }
+      }
+    } else if (value[i] == '\0') {
+      break;
+    } else {
+      new_value[j++] = value[i];
+    }
+  }
+
+  dt_conf_set_string("plugins/lighttable/export/resizing_factor", new_value);
+  gtk_entry_set_text (spin, new_value);
+}
+
+static void scale_changed_reset_sizes(GtkWidget *spin, dt_lib_export_t *d)
+{
+    dt_conf_set_int(CONFIG_PREFIX "height", 0);
+    g_signal_handler_block(G_OBJECT(d->height), d->height_chg1_handler_id);
+    g_signal_handler_block(G_OBJECT(d->height), d->height_chg2_handler_id);
+    gtk_entry_set_text(GTK_ENTRY(d->height), "0");
+    g_signal_handler_unblock(G_OBJECT(d->height), d->height_chg1_handler_id);
+    g_signal_handler_unblock(G_OBJECT(d->height), d->height_chg2_handler_id);
+
+    dt_conf_set_int(CONFIG_PREFIX "width", 0);
+    g_signal_handler_block(G_OBJECT(d->width), d->width_chg1_handler_id);
+    g_signal_handler_block(G_OBJECT(d->width), d->width_chg2_handler_id);
+    gtk_entry_set_text(GTK_ENTRY(d->width), "0");
+    g_signal_handler_unblock(G_OBJECT(d->width), d->width_chg1_handler_id);
+    g_signal_handler_unblock(G_OBJECT(d->width), d->width_chg2_handler_id);
+}
+
+static void size_changed_reset_scale(GtkWidget *spin, dt_lib_export_t *d)
+{
+  char new_value[2] = "1\0";
+  dt_conf_set_string(CONFIG_PREFIX "resizing_factor", new_value);
+  g_signal_handler_block(G_OBJECT(d->scale), d->scale_chg1_handler_id);
+  g_signal_handler_block(G_OBJECT(d->scale), d->scale_chg2_handler_id);
+  gtk_entry_set_text (GTK_ENTRY(d->scale), new_value);
+  g_signal_handler_unblock(G_OBJECT(d->scale), d->scale_chg1_handler_id);
+  g_signal_handler_unblock(G_OBJECT(d->scale), d->scale_chg2_handler_id);
+}
+
+static void scale_mdlclick(GtkEntry *spin, GdkEventButton *event, dt_lib_export_t *d)
+{
+  if (event->button == 2){
+    size_changed_reset_scale(GTK_WIDGET(spin), d);
+  } else {
+    scale_changed(spin);
+  }
+}
+
+static void widht_mdlclick(GtkEntry *spin, GdkEventButton *event, gpointer user_data)
```

Comment:
widht -> width

Also current dt style is to have all static routines to start with an underscore. I know there is some missing....